llvm/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -895,6 +895,21 @@ bool llvm::setLoopEstimatedTripCount(Loop *L, unsigned EstimatedTripCount,
   return true;
 }
 
+bool llvm::isInfiniteTripCount(Loop *L) {
+  if (BranchInst *ExitingBranch = getExpectedExitLoopLatchBranch(L)) {
+    uint64_t LoopWeight, ExitWeight;
```

Comment:
pet peeve: can you initialize the values? yes, they should be set by extractBranchWeights, but its implementation could, technically, decide at a point to leave an input unchanged, and uninitialized value bugs are not fun.